pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/Auth0Util.java

```diff
@@ -337,6 +337,41 @@ public Map<String, String> getAuth0UsersByEmails(Set<String> emailIds, String mg
         return results;
     }
 
+    public Map<String, String> getAuth0UsersByAuth0UserIds(Set<String> userIds, String mgmtApiToken) {
+        if (userIds == null || userIds.isEmpty()) {
+            return new HashMap<>();
+        }
+
+        int maxPerPage = 100;
+        Map<String, String> results = new HashMap<>();
+        List<String> ids = new ArrayList<>(userIds);
+        ManagementAPI auth0Mgmt = new ManagementAPI(baseUrl, mgmtApiToken);
+
+        for (int i = 0; i < ids.size(); i += maxPerPage) {
+            int end = Math.min(i + maxPerPage, ids.size());
+            List<String> subset = ids.subList(i, end);
+
+            String query = String.format("user_id:(%s)", String.join(" ", subset));
+
+            UserFilter filter = new UserFilter()
+                    .withFields("user_id,email", true)
+                    .withPage(0, 100)
+                    .withQuery(query)
+                    .withSearchEngine("v3");
+
+            try {
+                UsersPage page = auth0Mgmt.users().list(filter).execute();
+                for (User user : page.getItems()) {
+                    results.put(user.getId(), user.getEmail());
```

Comment:
zym
changed to set passed auth0Id as key rather than values and changed the CLI accordingly